### PR TITLE
Fix fields object in calendar.{date,yearMonth,monthDay}FromFields()

### PR DESCRIPTION
**Pull request by ptomato**
Closes: #905

### CODE REVIEW

**Comment by ptomato** [comment]
No accompanying spec text change, because the fromFields methods aren't in the spec yet. When we get to that point, we can refer to the polyfill for the logic.

File: polyfill/lib/ecmascript.mjs [hunk 1]
```diff
@@ -571,11 +571,8 @@ export const ES = ObjectAssign({}, ES2019, {
       const value = bag[property];
       if (value !== undefined) {
         any = any || {};
-        if (property === 'calendar') {
+        if (property === 'era') {
           // FIXME: this is terrible
```
Comment (ryzokuken) [inline polyfill/lib/ecmascript.mjs:575]: Still terrible? :smile:

File: polyfill/lib/ecmascript.mjs [hunk 2]
```diff
@@ -596,11 +593,8 @@ export const ES = ObjectAssign({}, ES2019, {
         }
         value = defaultValue;
       }
-      if (property === 'calendar') {
+      if (property === 'era') {
         // FIXME: this is terrible
```
Comment (ryzokuken) [inline polyfill/lib/ecmascript.mjs:597]: Same.

**Review by ryzokuken — APPROVED** [review]
LGTM, thanks. Please remove those two comments too?

> When we get to that point, we can refer to the polyfill for the logic.

Will do.

File: polyfill/lib/ecmascript.mjs (hunk 1 above)
Comment (ptomato) [reply polyfill/lib/ecmascript.mjs:575]: I still think it's terrible to special-case any property here, only slightly less so than before...

Comment (ryzokuken) [reply polyfill/lib/ecmascript.mjs:575]: Fair enough, will merge in that case.

**Pull request merged**